### PR TITLE
Profile wake startup times

### PR DESCRIPTION
**Pull request by V-FEXrt**
This is the first PR of several being split out from  https://github.com/sifive/wake/pull/1364

This change profiles wake startup time and dumps it into `wake.log` so that optimizations can be verified.

In the long term, it would be interesting to build out some kind of macro or C++ template for a `ProfiledStep` that accepts a low cost function and wraps in it profiling but I don't think that is needed just yet

### CODE REVIEW

File: tools/wake/main.cpp [hunk 1]
```diff
@@ -622,57 +639,64 @@ int main(int argc, char **argv) {
...
+
+    struct timeval starttv, stop;
+    gettimeofday(&starttv, 0);
+
+    for (size_t i = 0; i < wakefilenames.size(); i++) {
+      auto &wakefile = wakefilenames[i];
+
+      auto now = std::chrono::steady_clock::now();
+      if (!clo.quiet && is_stdout_tty &&
+          std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count() > 1000) {
```
Comment (JakeSiFive) [inline tools/wake/main.cpp:658]: nit: its very strange that we're using both chrono and gettimeofday

Comment (V-FEXrt) [reply tools/wake/main.cpp:658]: yeah, :/ this was lifted from some other profiling code we have. I can see about just using `chrono`

File: tools/wake/main.cpp [hunk 2]
```diff
@@ -594,8 +594,16 @@ int main(int argc, char **argv) {
...
-  auto wakefilenames =
-      find_all_wakefiles(enumok, clo.workspace, clo.verbose, libdir, ".", user_warn);
+  std::vector<std::string> wakefilenames;
+  {
+    auto start = std::chrono::steady_clock::now();
+    wakefilenames = find_all_wakefiles(enumok, clo.workspace, clo.verbose, libdir, ".", user_warn);
+    auto stop = std::chrono::steady_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count();
+    wcl::log::info("Find all wakefiles took %f seconds",
+                   static_cast<double>(duration) / 1000000.0)();
```
Comment (JakeSiFive) [inline tools/wake/main.cpp:604]: nit (And I stress this is a *nit*, you can land this as is):

```
    auto duration = std::chrono::duration_cast<std::chrono::seconds, double>(stop - start).count();
    wcl::log::info("Find all wakefiles took %f seconds", duration)();
```

File: tools/wake/main.cpp [hunk 3]
```diff
@@ -604,7 +612,15 @@ int main(int argc, char **argv) {
...
   Profile tree;
   Runtime runtime(clo.profile ? &tree : nullptr, clo.profileh, heap_factor);
-  bool sources = find_all_sources(runtime, clo.workspace);
+  bool sources = false;
+  {
+    auto start = std::chrono::steady_clock::now();
+    sources = find_all_sources(runtime, clo.workspace);
+    auto stop = std::chrono::steady_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::microseconds>(stop - start).count();
+    wcl::log::info("Find all sources took %f seconds", static_cast<double>(duration) / 1000000.0)();
```
Comment (JakeSiFive) [inline tools/wake/main.cpp:621]: ditto nit

**Review by JakeSiFive — APPROVED** [review]

File: tools/wake/main.cpp (hunk 2 above)
Comment (V-FEXrt) [reply tools/wake/main.cpp:604]: its trivial to  fix, and strictly better. No reason to not just fix it

**Pull request merged**